### PR TITLE
Add unit tests for namemunge and nameremove in devenv_impl

### PR DESCRIPTION
**Pull request by yungyuc**
@j8xixo12 @chestercheng I added a function for the main logic in `devenv_impl` to individually test `namemunge` and `nameremove` functions.

Issues #48 #49

### CODE REVIEW

File: scripts/env.d/devenv_impl [hunk 1]
```diff
@@ -17,24 +17,26 @@ nameremove() {
...
 
-case "$(uname)" in
-  Darwin)
-    lib_path=DYLD_FALLBACK_LIBRARY_PATH
-    ;;
-  Linux)
-    lib_path=LD_LIBRARY_PATH
-    ;;
-esac
+devenv_act() {
```
Comment (yungyuc) [inline scripts/env.d/devenv_impl:20]: The function that absorbs the main logic.

File: scripts/init [hunk 2]
```diff
@@ -4,6 +4,7 @@ export DEVENVROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 
 devenv() {
   . ${DEVENVROOT}/scripts/func.d/bash_utils
+  . ${DEVENVROOT}/scripts/env.d/devenv_impl
```
Comment (yungyuc) [inline scripts/init:7]: Reduce 3 sources into 1.

File: scripts/init [hunk 3]
```diff
@@ -14,10 +15,10 @@ devenv() {
     fi
 
     if [ ! -z "${DEVENVFLAVOR}" ] ; then
-      . ${DEVENVROOT}/scripts/env.d/devenv_impl nameremove ${DEVENVFLAVOR}
+      devenv_act nameremove ${DEVENVFLAVOR}
```
Comment (yungyuc) [inline scripts/init:18]: Call the main logic function.

File: tests/test_devenv_impl.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+test_namemunge() {
```
Comment (yungyuc) [inline tests/test_devenv_impl.sh:3]: Test for `namemunge`.

File: tests/test_devenv_impl.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  mypath=path1
+  assertEquals "$mypath" "path1"
+  . ${DEVENVROOT}/scripts/env.d/devenv_impl
+  namemunge mypath path2
+  assertEquals "$mypath" "path2:path1"
+  namemunge mypath path3 after
+  assertEquals "$mypath" "path2:path1:path3"
+}
+
+test_nameremove() {
```
Comment (yungyuc) [inline tests/test_devenv_impl.sh:13]: Test for `nameremove`.

**Review by tai271828 — APPROVED** [review]
Looks good to me.

**Comment by yungyuc** [comment]
> Looks good to me.

Thanks @tai271828 .

@chestercheng please let me know how do you think.

File: scripts/init (hunk 2 above)
Comment (chestercheng) [reply scripts/init:7]: It makes more effective!

**Comment by chestercheng** [comment]
LGTM

**Review by chestercheng — APPROVED** [review]

**Pull request merged**